### PR TITLE
docs: add Discord docs extlinks entry, update broken links

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Adds a `:ddocs:` role for easily linking to `https://discord.com/developers/docs/*`, and fixes a couple invalid links that are a result of api docs changes.

Also adds a `linkcheck_ignore` config for running the linkcheck builder without making unnecessary requests. This is meant to be run manually and hence I didn't integrate linkcheck into the ci workflow, mainly because (1) there really isn't a point in doing this every time, and (2) checking fragments/anchors for api docs isn't possible since the entire page is generated dynamically.

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
Thoughts on naming the role `ddocs` instead?

**Comment by shiftinv** [comment]
> Thoughts on naming the role `ddocs` or `ddevs` instead?

Sounds good, done in https://github.com/DisnakeDev/disnake/pull/793/commits/165d9bfb58488a4de4ab48bda3c692b95b75e2ed.

File: docs/conf.py [hunk 1]
```diff
@@ -235,6 +236,12 @@ def linkcode_resolve(domain: str, info: Dict[str, Any]) -> Optional[str]:
     hoverxref_api_host = "/_"
 
 
+linkcheck_ignore = [
+    r"https?://github.com/.+?/.+?/(issues|pull)/\d+",
```
Comment (onerandomusername) [inline docs/conf.py:240]: why are we ignoring these?

Comment (shiftinv) [reply docs/conf.py:240]: Ignoring these avoids around ~650 requests to Github, which are fairly unnecessary and get you ratelimited rather quickly; and the `support.discord.com` links consistently return a 403 here thanks to Cloudflare.

**Review by onerandomusername — APPROVED** [review]

**Pull request merged**